System-Testing-Deployments-and-Upgrades/User-Authorization/index.rst

```diff
@@ -0,0 +1,162 @@
+For example, the control classes have information about the components it interacts with and, therefore, can automatically produce a list of the CSCs that need requests.
+They can also determine the correct user identity without the user having to know where to get it from.
+
+The request is done via the ``Authorize`` CSC, which parses the request, handle communication with ``LOVE`` to authorize with an administrator and sets the authorizations in the CSCs.
+
+The following is an example of how to request authorization to command the ``MTRotator`` CSC.
+
+.. code-block:: python
+
+  from lsst.ts import salobj
+
+  domain = salobj.Domain()
+
+  # As mentioned above, we need to request the authorization to the Authorize CSC
+  # so we must construct a remote to send the command to it.
+  authorize = salobj.Remote(domain, "Authorize")
+
+  await authorize.start_task
+
+  identity = domain.user_host
+  timeout = 60.0
+
+  # Request authorization to command MTRotator
+  await authorize.cmd_requestAuthorization.set_start(
+    cscsToChange="MTRotator",
+    authorizedUsers=f"+{identity}",
+    timeout=timeout
+  )
+
+  # User is now authorized to command the MTRotator CSC
+  ...
+
+  # Release authorization to command MTRotator
+  await authorize.cmd_requestAuthorization.set_start(
+    cscsToChange="MTRotator",
+    authorizedUsers=f"-{identity}",
+    timeout=timeout
+  )
+
+.. _User-Authorization-Troubleshooting:
+
+Troubleshooting
+===============
+
+Failure to Set Authorization
+----------------------------
+
+One common problem users may face in requesting authorization failure to set authorization by the ``Authorize`` CSC.
+The error message will look like the following:
+  
+  AckError: msg='Command failed', ackcmd=(ackcmd private_seqNum=1003719146, ack=<SalRetCode.CMD_FAILED: -302>, error=1, result="Failed: Failed to set authList for one or more CSCs: **{'ATPtg': 'Timed out waiting for command acknowledgement'}**. The following CSCs were successfully updated: {'ATHexapod', 'ATPneumatics', 'ATDomeTrajectory', 'ATAOS', 'ATMCS', 'ATDome'}.")
+
+The relevant information is highlighted in bold letters.
+In the example above, The ``ATPtg`` component timed out replying to the set authorization command from the ``Authorize`` CSC.
+In this case, this happened because authorization was not enabled in the component.
+
+Authorization Request Completed But Not Authorized
+--------------------------------------------------
+
+One thing to keep in mind about the interaction between the ``Authorize`` CSC and ``LOVE`` is that the CSC returns as soon as LOVE acknowledges that the request was sent to the administrator.
+This does not mean the authorization was set successfully, only that the administrator should have received the request.
+The authorization will only be set after the administrator approves it in the frontend.
```

Comment:
That's not obvious.
Is there a list/GUI etc that says which person is authorized to use each CSC?

Reply:
Uhm, yes... this is mentioned in line 77 with an accompanying figure.

Reply:
Ya I missed that.  
The fact that the UI is basically empty of information makes it more difficult to get an idea of how it actually works. 
I'm assuming that the LOVE bit isn't developed enough to show it actually being used?

Reply:
bah, there is that info but on a different page... 
hrmph. 
I think the high level issue is again my lack of understanding, combined with the fact that there isn't much explanation of how this works.
I do think that just showing the GUI without anyone using a CSC doesn't offer much.

Reply:
yeah... the views are pretty bare at the moment.. hopefully they will improve once we start using it..

Reply:
So basically it's expected that users open the auth list and manually monitor that they've been approved? There is no way to query the system and verify they have been authorized?

Reply:
yes, you can check the authorization.. Maybe I should add some utility functions to the high level package..

Reply:
Ok, I included some content about checking authorization from Nublado.